check.go

```diff
@@ -1,32 +1,67 @@
 package filenameslinter
 
 import (
+	"errors"
 	"fmt"
+	"io/fs"
 	"path"
-    "io/fs"
-    "regexp"
+	"regexp"
+	"sort"
 
 	"github.com/csunibo/synta"
 	syntaRegexp "github.com/csunibo/synta/regexp"
 )
 
-func CheckDir(synta synta.Synta, fs fs.ReadDirFS, dirPath string) (err error) {
-    entries, err := fs.ReadDir(dirPath)
+func CustomReadDir(fsys fs.FS, name string) ([]fs.DirEntry, error) {
+	if fsys, ok := fsys.(fs.ReadDirFS); ok {
+		return fsys.ReadDir(name)
+	}
+
+	file, err := fsys.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	dir, ok := file.(fs.ReadDirFile)
+	if !ok {
+		return nil, &fs.PathError{Op: "readdir", Path: name, Err: errors.New("not implemented")}
+	}
+
+	list, err := dir.ReadDir(-1)
+	sort.Slice(list, func(i, j int) bool { return list[i].Name() < list[j].Name() })
+	return list, err
+}
+
+func CheckDir(synta synta.Synta, fs fs.FS, dirPath string, recursive bool, kebab_dir bool) (err error) {
+	entries, err := CustomReadDir(fs, dirPath)
 	if err != nil {
 		return
 	}
 
 	for _, entry := range entries {
 		file, err := entry.Info()
 		if err != nil {
-            err = fmt.Errorf("Could not read directory: %v", err)
+			err = fmt.Errorf("Could not read directory: %v", err)
 			return err
 		}
 
 		if file.IsDir() {
-			if err := CheckName(synta, file.Name(), true); err != nil {
-				dirPath = path.Join(dirPath, file.Name())
-				if err := CheckDir(synta, fs, dirPath); err != nil {
+			kebab_regexp := regexp.MustCompile("^[a-z0-9]+(-[a-z0-9]+)*$")
```

Comment:
Mettiamolo in una costante fuori dalla funzione.

Reply:
L'ho messa dentro una `var`, la costante non e' possibile visto che restituisce un puntatore (credo)